internal/cephfs/volumeoptions.go

```diff
@@ -299,7 +301,15 @@ func newVolumeOptionsFromVolID(ctx context.Context, volID string, volOpt, secret
 	}
 
 	volOptions.ProvisionVolume = true
-	volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+	info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+	if err != nil {
```

Comment:
multiple if else if can be converted as below?

```go
info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
if err == nil {
 volOptions.RootPath = info.Path
 volOptions.Features = info.Features
}

if err != nil && errors.Is(err, ErrInvalidCommand) {
volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
}

return &volOptions, &vid, err
```

Reply:
we also want to return VolumeOptions for successful case, which brings us to this :
```
info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
if err == nil {
 volOptions.RootPath = info.Path
 volOptions.Features = info.Features
}

if err != nil && errors.Is(err, ErrInvalidCommand) {
volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
}
if err != nil {
	return &volOptions, &vid, err
}
	return &volOptions, &vid, nil
```
This will result in nearly similar situation we have now. Should I change it?

Reply:
```go
// If error other than ErrInvalidCommand
if err != nil {
	return &volOptions, &vid, err
}
```
volumeOptions are returned in all the cases why this check is required?

Reply:
Sorry, I am a bit confused. In https://github.com/ceph/ceph-csi/pull/1458#discussion_r486218737 , How are we returning the success case i.e `return &volOptions, &vid, nil` . Please correct me if I am missing something

Reply:
err will be nil if `getSubVolumeInfo` succeeded or if it's an ErrInvalidCommand it will be overridden by `getVolumeRootPathCeph` so `err` will be set to `nil` in both success cases.

Reply:
Oh Got it. Thanks for explaining :)